src/Main.elm

```diff
@@ -1,10 +1,571 @@
+    Zipper.fromTree (Maybe.withDefault defaultTree tree)
+        |> Zipper.findFromRoot (\id -> id == node)
+        |> Maybe.andThen Zipper.removeTree
+        |> Maybe.map Zipper.toTree
+
+
+defaultTree : Tree.Tree Node
+defaultTree =
+    Tree.singleton (File defaultAudience.id defaultAudience)
+
+
+allNodesInDict : Dict Id Node
+allNodesInDict =
+    let
+        getId : Node -> Id
+        getId node =
+            case node of
+                Dir id audienceFolder ->
+                    id
+
+                File id audience ->
+                    id
+
+                Root ->
+                    0
+    in
+    allNodes
+        |> List.map (\node -> ( getId node, node ))
+        |> Dict.fromList
+
+
+allNodes : List Node
+allNodes =
+    let
+        audienceFolders : List AudienceFolder
+        audienceFolders =
+            Maybe.withDefault [ defaultAudienceFolder ] decodeAudienceFolders
+
+        audiences : List Audience
+        audiences =
+            Maybe.withDefault [ defaultAudience ] decodeAudiences
+    in
+    List.append
+        (List.map (\audienceFolder -> Dir audienceFolder.id audienceFolder) audienceFolders)
+        (List.map (\audience -> File audience.id audience) audiences)
+
+
+
+-- Utility functions for traversing a Tree
+-- Finding a Tree with node
+-- Get children Nodes of certain parent tree
+
+
+findATree : Node -> Tree Node -> Maybe (Zipper.Zipper Node)
+findATree node tree =
+    Zipper.fromTree tree
+        |> Zipper.findFromRoot (\id -> id == node)
+
+
+getChildrenNodes : Tree Node -> List Node
+getChildrenNodes tree =
+    tree
+        |> Tree.children
+        |> List.map Tree.label
+
+
+
+-- UPDATE
+
+
+type Msg
+    = NoOp
+    | GoTo Node
+    | GoUp
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        NoOp ->
+            ( model, Cmd.none )
+
+        GoTo node ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    findATree node model.directory
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+        GoUp ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    model.zipperStep
+                        |> Maybe.andThen Zipper.parent
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+
+
+-- VIEW
+
+
+type alias Model =
+    { directory : Tree Node
+    , currentTree : Maybe (Tree Node)
+    , zipperStep : Maybe (Zipper.Zipper Node)
```

Comment:
What's the reason for keeping multiple representations of the same data? (If you have a Zipper, you have the whole tree. `Zipper.fromTree : Tree a -> Zipper a` and `Zipper.toTree : Zipper a -> Tree a` allow you to convert back and forth without loss of information, if you need to.)